src/index.js

```diff
@@ -46,25 +46,32 @@ function runSync (command, options) {
 }
 
 function runAsync (command, options) {
-  // Prepare options for exec command (don't need async and stdio as it doesn't handle them)
-  const execOptions = Object.assign({}, options)
-  delete execOptions.async
-  delete execOptions.stdio
-
   return new Promise((resolve, reject) => {
-    const asyncProcess = childProcess.exec(command, execOptions, (error, stdout) => {
-      if (error) {
-        reject(error)
-      } else {
-        resolve(stdout.toString())
+    const spawnOptions = Object.assign({shell: true}, options)
+    const timeout = spawnOptions.timeout
+    delete spawnOptions.async
+
+    return new Promise((resolve, reject) => {
+      const asyncProcess = childProcess.spawn(command, spawnOptions)
+      asyncProcess.on('error', (error) => {
+        reject(new Error(`Failed to start command: ${command}; ${error}`))
+      })
+
+      asyncProcess.on('close', (exitCode) => {
+        if (exitCode === 0) {
+          resolve(exitCode)
```

Comment:
One problem I was running into was that `asyncProcess.stdout` was null. I think that's because stdio was `inherit` but if I would change it from that it wouldn't actually do the streaming. I tried a few different things, but couldn't get the right combo of options. Since I don't have a good grasp of how things are expected to act when different stdio options are used, I'll leave the buffer piece.
However, that means there's still a failing unit test.